login_server/src/cpp/model/email/Email.cpp

```diff
@@ -156,6 +156,7 @@ Gradido Login-Server\n\
 		auto em = EmailManager::getInstance();
 		auto adminRecipient = Net::MailRecipient(Net::MailRecipient::PRIMARY_RECIPIENT, em->getAdminReceiver());
 		Poco::AutoPtr<model::table::User> userTableModel;
+		mResendCounter++;
```

Comment:
This is probably the most problematic code section - you count up the resend count at a place where neither the sending nor resending of EMails takes place - its a different file tho.
I understand that this makes it quite easy regarding lines of codes to be changed, but it also can have sie effects. Are you sure there will be 5 tries to send the EMail or will some/all emails call the draft method when creating the email and therefore start with `mResendCounter == 1`.
I would expect the counter to go up where the resend actually is qued.

Reply:
I found it always hard to understand code changes only from the diff, I need the whole picture.
When you look at the whole picture your understand. 
You know I work object oriented with c++.
Mails are handled as own objects, if some part of the programm likes to send a email it creates an email object, add every needed data to it which is needed to create the actual email and push it to the email manager queue. 

The Email Manager get the first email from the queue, call draft and try to send the email. 
If the connection failed, now the email will be put back into the queue.

And yes I have checked that draft is only called from EmailManager.

If I would call mResendCounter++ after or before push I must put it into two places (double code), or I must call inside the EmailManager::ThreadFunction() the addEmail() function which will call condSignal() {condition.signal() }. 
I don't know the side effects from calling condition signal inside the thread which will be waiting on the condition.